datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -0,0 +1,79 @@
+package com.ds.datastore;
+
+
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RestController
+public class BookController {
+
+    private final BookRepository repository;
+
+    private final BookModelAssembler assembler;
+
+    BookController(BookRepository repository, BookModelAssembler assembler) {
+
+        this.repository = repository;
+        this.assembler = assembler;
+    }
+
+    @PostMapping("/books")
+    Book newBook(@RequestBody Book book) {
```

Comment:
I believe all these methods should be marked `private`. In general, best practice is to have the most restrictive access possible.